mediaflow_proxy/extractors/base.py

```
@@ -0,0 +1,42 @@
+            async with httpx.AsyncClient(proxy=self.proxy_url) as client:
+                response = await client.get(
+                    url,
+                    headers={**self.base_headers, **(headers or {})},
+                    follow_redirects=follow_redirects,
+                    timeout=30,
+                    **kwargs,
+                )
+                response.raise_for_status()
+                return response
+        except httpx.HTTPError as e:
+            raise ValueError(f"HTTP request failed: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)